cookiecutter/prompt.py

```diff
@@ -274,7 +274,10 @@ def prompt_choice_for_config(
     """
     rendered_options = [render_variable(env, raw, cookiecutter_dict) for raw in options]
     if no_input:
-        return rendered_options[0]
+        if len(rendered_options) != 0:
+            return rendered_options[0]
+        else:
+            return None
     return read_user_choice(key, rendered_options, prompts, prefix)
```

Comment:
This cannot return `None` -- that contradicts the return type annotation (`OrderedDict` or `str`).

This should be formulated as:

```python
    if not rendered_options:
        raise ValueError
    if no_input:
        return rendered_options[0]
    return read_user_choice(key, rendered_options, prompts, prefix)
```

`read_user_choice()` has a condition to ensure that the list of options isn't empty, but we can catch that here.

This also needs to have a test added to the test suite so that this issue is not simply fixed, but _provably and permanently_ fixed.